scripts/generateIcons.js

```diff
@@ -0,0 +1,197 @@
+    { removeViewBox: false },
+    {
+      inlineStyles: {
+        onlyMatchedOnce: false,
+      },
+    },
+    { convertStyleToAttrs: true },
+  ],
+});
+
+const svgrConfig = {
+  svgProps: { fill: 'currentColor', width: 16, height: 16 },
+  replaceAttrValues: { '#000': 'currentColor' },
+  template: componentTemplate,
+  plugins: ['@svgr/plugin-jsx', '@svgr/plugin-prettier'],
+};
+
+const baseDir = path.join(__dirname, '..');
+const iconComponentsDir = path.join(baseDir, 'lib/components/icons');
+
+(async () => {
+  // First clean up any existing SVG components
+  const existingComponentPaths = await globby(
+    'lib/components/icons/*/*Svg.tsx',
+    {
+      cwd: baseDir,
+      absolute: true,
+    },
+  );
+  await Promise.all(
+    existingComponentPaths.map(async existingComponentPath => {
+      await fs.remove(existingComponentPath);
+    }),
+  );
+
+  // Load SVGs
+  const svgFilePaths = await globby('icons/*.svg', {
+    cwd: baseDir,
+    absolute: true,
+  });
+
+  await Promise.all(
+    svgFilePaths.map(async svgFilePath => {
+      // Split out the icon variants (e.g. bookmark-active.svg)
+      const [svgName, variantName] = path
+        .basename(svgFilePath, '.svg')
+        .split('-');
+
+      const rawSvg = await fs.readFile(svgFilePath, 'utf-8');
+      const svg = rawSvg.replace(/ data-name=".*?"/g, '');
+
+      // Run through SVGO
+      const optimisedSvg = (await svgo.optimize(svg)).data;
+
+      // Validate SVG before import
+      const $ = cheerio.load(optimisedSvg);
+      $('svg *').each((i, el) => {
+        const $el = $(el);
+
+        // Validate color attributes
+        ['stroke', 'fill'].forEach(attr => {
+          const color = $el.attr(attr);
+          const validColors = ['currentColor', 'black', '#000', '#000000'];
+          if (color && !validColors.includes(color)) {
+            throw new Error(`${svgName}: Invalid ${attr} color: ${$.html(el)}`);
+          }
+        });
+      });
+
+      const iconName = `${pascalCase(svgName)}Icon`;
+      const svgComponentName = `${iconName}${
+        variantName ? pascalCase(variantName) : ''
+      }Svg`;
+      const svgComponent = await svgr(optimisedSvg, svgrConfig, {
+        componentName: svgComponentName,
+      });
+
+      // Create icon directory if it's missing
+      const iconDir = path.join(iconComponentsDir, iconName);
+      await fs.mkdirp(iconDir);
+
+      // Write SVG React component
+      await fs.writeFile(
+        path.join(iconDir, `${svgComponentName}.tsx`),
+        svgComponent,
+        { encoding: 'utf-8' },
+      );
+
+      // Bail out early if we're processing an icon variant (e.g. bookmark-active.svg)
+      // All subsequent steps should only happen once per icon component.
+      if (variantName) {
+        return;
+      }
+
+      const templateFileIfMissing = async (relativePath, contents) => {
+        const filePath = path.join(iconDir, relativePath);
+        if (!(await fs.exists(filePath))) {
```

Comment:
Just looked into this and `fs-extra` provide the `fs.exists` api even though it's deprecated in node. They also export a first class api called [`pathExists`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md), which is probably a safer option in that it's less likely to get deprecated.